rcl/src/rcl/subscription.c

```diff
@@ -100,8 +100,11 @@ rcl_subscription_init(
     1, sizeof(rcl_subscription_impl_t), allocator->state);
   RCL_CHECK_FOR_NULL_WITH_MSG(
     subscription->impl, "allocating memory failed", ret = RCL_RET_BAD_ALLOC; goto cleanup);
+
+  // options
+  subscription->impl->options = *options;
```

Comment:
i think originally impl structure is created after `rmw_handle` is created successfully. if we take this change, maybe the same change can be added to `rcl_publisher_init`?

Reply:
I moved this earlier because in the case that `rmw_create_subscription` fails, we try to run `rcl_subscription_options_fini` on the uninitialized structure, which returns its own warning for an uninitialized structure.  By copying it earlier, we avoid that in cleanup paths.

For the publisher, we have no equivalent of `rcl_subscription_options_fini`; it is a POD type.  So when we copy it doesn't really matter.  I don't have a strong opinion either way, to be honest.

Reply:
i see. i think that is okay with current code.